keyboards/keycapsss/plaid_pad/keymaps/via/keymap.c

```diff
@@ -0,0 +1,99 @@
+/* Copyright 2020 KEYCAPSSS
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+/*
+ * ,-----------------------,
+ * |  7  |  8  |  9  |  /  |
+ * |-----+-----+-----+-----|
+ * |  4  |  5  |  6  |  *  |
+ * |-----+-----+-----+-----|
+ * |  1  |  2  |  3  |  -  |
+ * |-----+-----+-----+-----|
+ * |  0  |  .  |  =  |  +  |
+ * `-----------------------'
+ */
+  [0] = LAYOUT_ortho_4x4(
+      KC_P7,   KC_P8,   KC_P9,   KC_PSLS,
+      KC_P4,   KC_P5,   KC_P6,   KC_PAST,
+      KC_P1,   KC_P2,   KC_P3,   KC_PMNS,
+      KC_P0,   KC_PDOT, KC_PEQL, KC_PPLS  ),
+  [1] = LAYOUT_ortho_4x4(
+      KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+      KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+      KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+      KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS  ),
+  [2] = LAYOUT_ortho_4x4(
+      KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+      KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+      KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+      KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS  ),
+  [3] = LAYOUT_ortho_4x4(
+      KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+      KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+      KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+      KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS  ),
+};
+
+// Set LED1 state during powerup
+void keyboard_post_init_user(void) {
+    writePinHigh(LED_RED);
+}
+
+void encoder_update_user(uint8_t index, bool clockwise) {
+/*
+ * ,-----------------------,
+ * |  E1 |  E2 |  E3 |  E4 |
+ * |-----+-----+-----+-----|
+ * |     |     |     |  E3 |
+ * |-----+-----+-----+-----|
+ * |     |     |     |  E2 |
+ * |-----+-----+-----+-----|
+ * |     |     |     |  E1 |
+ * `-----------------------'
+ */
+
+  // First encoder
+  if (index == 0) {
```

Comment:
There are only two encoder spots on this board. Is there a new revision?

Reply:
Hello,
yes i have a R1.1 of the Plaid-Pad pcb.
I forgot to add the extra encoder pads to the config.h
Fixed with a0f04ea9a361207b0272286d63f05548e3795bbe

Reply:
Hm, I'm not sure but this may need to be in a separate revision folder, because the placement of the second encoder has changed. I'll let the other collabs weigh in.

Reply:
The code works with Rev1 (2 encoder) and Rev1.1 (4 encoder).
The encoder are marked on the pcb's.

Reply:
Yes, however for example I have my second encoder set to scroll up and down, and the key at that position as left click - this would not really make sense on the new revision since the left click would then be on the fourth encoder, but again, I'm not entirely sure how valid that is of an argument.

Reply:
Ok i understand. Thank you for the explanation.

Reply:
After thinking about it, IMO this board definitely should be split up into `rev1` and `rev1_1`, due to the two extra encoders, and the fact that the second encoder by index has changed position.

Reply:
@drashna wrote it's fine.

Reply:
@fauxpark i updated the default keymap and the via and added more informations about the encoders.
Also updated the readme's.

Reply:
@fauxpark added rev folders and tested the firmware on rev1 and rev1.1 pcb's.